src/rules/fileNameCasingRule.ts

```diff
@@ -25,47 +25,95 @@ enum Casing {
-            * \`${Casing.SnakeCase}\`: File names must be snake-cased: \`file_name.ts\`.`,
+            * In both cases the casing string must be one of the options:
+            ** \`${Casing.CamelCase}\`: File names must be camel-cased: \`fileName.ts\`.
+            ** \`${Casing.PascalCase}\`: File names must be pascal-cased: \`FileName.ts\`.
+            ** \`${Casing.KebabCase}\`: File names must be kebab-cased: \`file-name.ts\`.
+            ** \`${Casing.SnakeCase}\`: File names must be snake-cased: \`file_name.ts\`.
+        
+            * The array again consists of array with two items. The first item must be a case-insenstive
+            regular expression to match files, the second item must be a valid casing option (see above)`,
         options: {
-            type: "array",
-            items: [
-                {
-                    type: "string",
-                    enum: [Casing.CamelCase, Casing.PascalCase, Casing.KebabCase, Casing.SnakeCase],
-                },
-            ],
+            type: "list",
+            listType: {
+                anyOf: [
+                    {
+                        type: "string",
+                        enum: [
+                            Casing.CamelCase,
+                            Casing.PascalCase,
+                            Casing.KebabCase,
+                            Casing.SnakeCase
+                        ]
+                    },
+                    {
+                        type: "array",
+                        items: [
+                            {
+                                type: "string"
+                            },
+                            {
+                                type: "string",
+                                enum: [
+                                    Casing.CamelCase,
+                                    Casing.PascalCase,
+                                    Casing.KebabCase,
+                                    Casing.SnakeCase
+                                ]
+                            }
+                        ]
+                    }
+                ]
+            }
         },
         optionExamples: [
             [true, Casing.CamelCase],
             [true, Casing.PascalCase],
             [true, Casing.KebabCase],
             [true, Casing.SnakeCase],
+            [
+                true,
+                [
+                    [".style.ts$", Casing.KebabCase],
+                    [".tsx$", Casing.PascalCase],
+                    [".*", Casing.CamelCase]
+                ]
+            ]
         ],
         hasFix: false,
         type: "style",
-        typescriptOnly: false,
+        typescriptOnly: false
     };
     /* tslint:enable:object-literal-sort-keys */
 
     private static FAILURE_STRING(expectedCasing: Casing): string {
         return `File name must be ${Rule.stylizedNameForCasing(expectedCasing)}`;
     }
 
+    private static isValidCasingOption(casing: string) {
```

Comment:
Nit: instead of this dedicated function, you can keep a `const validCasingOptions = new Set([Casing.CamelCase, ...]);` outside of the class and use its `.has`. A little less code.